app/controllers/api/doctors_controller.rb

```diff
@@ -0,0 +1,39 @@
+class Api::DoctorsController < ApplicationController
+  def index
+    @doctors = Doctor.all
+    render json: @doctors,
+           only: %i[id name specialty years_of_experience bio age qualifications location_of_work photo]
+  end
+
+  def destroy
+    @doctor = Doctor.find_by(id: params[:id])
+    if @doctor
+      @doctor.transaction do
+        @doctors_reservations = DoctorsReservation.where(doctor_id: @doctor.id)
+        @reservations = Reservation.where(id: @doctors_reservations.map(&:reservation_id))
+        @reservations.destroy_all # deletes all reservations of the doctor
+        @doctors_reservations.destroy_all # deletes all records linking the doctor and reservations
```

Comment:
- [ ] Kindly, remove this old commented code in order to improve the repo's cleanliness. 👍